orangecontrib/text/widgets/owdocumentembedding.py

```diff
@@ -0,0 +1,234 @@
+    aggregator = Setting(default='mean')
+
+    def __init__(self):
+        OWWidget.__init__(self)
+        ConcurrentWidgetMixin.__init__(self)
+
+        self.languages = list(LANGS_TO_ISO.keys())
+        self.aggregators = AGGREGATORS
+        self.corpus = None
+        self.new_corpus = None
+        self._setup_layout()
+
+    def _setup_layout(self):
+        self.controlArea.setMinimumWidth(self.controlArea.sizeHint().width())
+
+        widget_box = widgetBox(self.controlArea, 'Settings')
+
+        self.language_cb = comboBox(widget=widget_box,
+                                    master=self,
+                                    value='language',
+                                    label='Language: ',
+                                    orientation=Qt.Horizontal,
+                                    items=self.languages,
+                                    callback=self._option_changed)
+
+        self.aggregator_cb = comboBox(widget=widget_box,
+                                      master=self,
+                                      value='aggregator',
+                                      label='Aggregator: ',
+                                      orientation=Qt.Horizontal,
+                                      items=self.aggregators,
+                                      callback=self._option_changed)
+
+        self.auto_commit_widget = auto_commit(widget=self.controlArea,
+                                              master=self,
+                                              value='_auto_apply',
+                                              label='Apply',
+                                              commit=self.commit)
+
+        self.cancel_button = QPushButton('Cancel',
+                                         icon=self.style()
+                                         .standardIcon(QStyle.SP_DialogCancelButton))
+
+        self.cancel_button.clicked.connect(self.cancel)
+
+        hbox = hBox(self.controlArea)
+        hbox.layout().addWidget(self.cancel_button)
+        self.cancel_button.setDisabled(True)
+
+    def set_input_corpus_summary(self, corpus):
+        if corpus is None:
+            self.info.set_input_summary(self.info.NoInput)
+        else:
+            self.info.set_input_summary(str(len(corpus)), "{} documents."
+                                        .format(len(corpus)))
+
+    def set_output_corpus_summary(self, corpus):
+        if corpus is None:
+            self.info.set_output_summary(self.info.NoOutput)
+        else:
+            unsuccessful = np.sum(np.isnan(corpus.X)) / 300
+            if unsuccessful > 0:
+                self.Warning.unsuccessful_embeddings()
+            self.info.set_output_summary(str(int(len(corpus) - unsuccessful)),
+                                         "Successful: {}, Unsuccessful: {}"
+                                         .format(int(len(corpus) - unsuccessful),
+                                                 int(unsuccessful)))
```

Comment:
I know it is not fitting in one line but here I would suggest different formating that would make this more readable.
```
self.info.set_output_summary(
    str(int(len(corpus) - unsuccessful)),
    "Successful: {}, Unsuccessful: {}".format(int(len(corpus) - unsuccessful),  int(unsuccessful))
)
```
If it does not fit in the line it is ussually prefearable to break after first ( and in this case when both parameters are long to put each in a seperate line. If the second parametere still does not fit break it after `format(`